hbook/ops.go

```diff
@@ -11,7 +11,8 @@ import (
 
 // DivideH1D divides 2 1D-histograms and returns a 2D scatter.
 // DivideH1D returns an error if the binning of the 1D histograms are not compatible.
-func DivideH1D(num, den *H1D) (*S2D, error) {
+// DivideH1D ignores points with NaN values if ignoreNaN is true.
+func DivideH1D(num, den *H1D, ignoreNaN bool) (*S2D, error) {
```

Comment:
actually, I think I'd rather use the "functional option" pattern [1,2].

with having a default for `DivOptions` that matches the current behaviour (keeping the NaNs).
I could imagine the following definitions:
```go
func DivideH1D(num, den *H1D, opts ...DivOptions) (*S2D, error) {
    cfg := newDivConfig()
    for _, opt := range opts {
        opt(cfg)
    }

    // ... ignore if cfg.ignoreNaNs, use cfg.replace() otherwise ...
}

type DivOptions func(c *divConfig)
type divConfig struct {
    ignoreNaNs bool
    replace func() float64
}

func newDivConfig() *divConfig {
    return &divConfig{replace: func() float64 { return math.NaN() }}
}

func DivIgnoreNaNs() DivOptions {
    return func(c *divConfig) {
        c.ignoreNaNs = true
    }
}

func DivReplaceNaNs(v float64) DivOptions {
    return func(c *divConfig) {
        c.replace = func() float64{ return v }
    }
}
```

it's a fair amount of boilerplate code, granted.
but neat.

do you want to try ironing it out?

[1]: https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis
[2]: https://godoc.org/go-hep.org/x/hep/hplot#FigOption

Reply:
OK, yes this is nice. I need to think about the logic (not too used to it). Let me give a shot and we'll see if I manage to do it.